workers/generic-worker/multiuser_test.go

```diff
@@ -56,3 +60,53 @@ func TestWhoAmI(t *testing.T) {
 
 	_ = submitAndAssert(t, td, payload, "completed", "completed")
 }
+
+func TestPrivilegedGenericWorkerBinaryFailsWorker(t *testing.T) {
+	setup(t)
+
+	if config.RunTasksAsCurrentUser {
+		t.Skip("Skipping since we're testing if the generic-worker binary is executable by the task user.")
+	}
+
+	goPath, err := host.CombinedOutput("go", "env", "GOPATH")
+	if err != nil {
+		t.Fatalf("Could not get GOPATH: %v", err)
+	}
+	if goPath == "" {
+		t.Fatal("GOPATH is empty")
+	}
+	goPath = strings.TrimSpace(goPath)
+
+	permissionsBeforeStr, err := fileutil.GetPermissionsString(goPath)
+	if err != nil {
+		t.Fatalf("Could not get permissions (string) of GOPATH: %v", err)
+	}
+	var fileMode fs.FileMode
+	if runtime.GOOS != "windows" {
+		fileMode, err = fileutil.GetPermissions(goPath)
+		if err != nil {
+			t.Fatalf("Could not get permissions of GOPATH: %v", err)
+		}
+	}
```

Comment:
Could GetPermissions return a String representation of the permissions _and_ a function that resets the permissions back to these values? Then you wouldn't need to special case Windows and have both the GetPermissions and GetPermissionsString versions, you would just have a single function that returned a string and a reset function.